### PR TITLE
Hashes need to be checked when looking for updates

### PR DESCRIPTION
**Pull request by DavidTanner**

### CODE REVIEW

**Comment by skparkes** [comment]
We just got hit with this nasty bug too.  Unfortunately, it started building from pull request ~3 of 2,700, so we had to go ahead and turn off CI until this gets fixed.  Thanks for working on it!

**Comment by DavidTanner** [comment]
Can you guys post some logs so I fix the right thing

**Comment by skparkes** [comment]
I'm going to get my colleague to try and pull the appropriate logs for you.  I believe he'll be posting in here shortly to confirm which logs would be the most helpful.

**Comment by sgredell** [comment]
@DavidTanner are you looking for "console output" on one of the runs we are having a problem with?

**Comment by sgredell** [comment]
Here is the console output for one of the failing CI tests (with much of the middle removed so I could actually upload it somewhere):
[jenkins_dump.txt](https://github.com/jenkinsci/ghprb-plugin/files/280300/jenkins_dump.txt)

**Comment by DavidTanner** [comment]
That log doesn't look like the plugin started it.  It was started by a different trigger.  The log I am interested in is the main jenkins log where the plugin says what it is doing.  Check the config to make sure the only trigger is the Pull Request builder.

**Comment by sgredell** [comment]
Sorry here's the syslog right after a pull request was started:

```
May 25, 2016 3:27:52 PM INFO org.jenkinsci.plugins.ghprb.GhprbRootAction handleAction
Checking PR #2,790 for PRM/analytics-pipeline
May 25, 2016 3:27:52 PM INFO org.jenkinsci.plugins.ghprb.GhprbGitHubAuth checkSignature
Signatures checking OK
May 25, 2016 3:27:52 PM INFO org.jenkinsci.plugins.ghprb.GhprbTrigger handlePR
Checking PR #2,790 for job CI_PRM
May 25, 2016 3:27:52 PM INFO org.jenkinsci.plugins.ghprb.GhprbPullRequest 
Created Pull Request #2,790 on PRM/analytics-pipeline by steve-gredell () updated at: 5/25/16 3:27 PM SHA: 756edaad180528adb912ac076ebde1ba05a11c02
May 25, 2016 3:27:52 PM INFO org.jenkinsci.plugins.ghprb.GhprbPullRequest updatePR
Pull request #2,790 was updated on repo PRM/analytics-pipeline but there aren't any new comments nor commits; that may mean that commit status was updated.
May 25, 2016 3:28:40 PM INFO hudson.model.Run execute
CI_PRM #3621 main build action completed: FAILURE
```

Let me know if you need anything else.  Thanks.

**Comment by marcmillien** [comment]
Thanks for working on this :).

We are also impacted here and I am asking myself if downgrading the plugin could solves the issue and so to which version we should downgrade for it.

Any of you tried this ? I plan to try it tomorrow

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 1]
```diff
@@ -187,88 +187,105 @@ public void check(GHIssueComment comment) {
...
+                // Update the PR object with the new pull request object if
+                // it is non-null. getPullRequest will then avoid another
+                // GH API call.
+                if (ghpr != null) {
+                    setPullRequest(ghpr);
+                }
+
+                Date lastUpdateTime = updated;
+                Date updatedDate = comment != null ? comment.getUpdatedAt() : ghpr.getUpdatedAt();
+                boolean wasUpdated = isUpdated(getPullRequest(), updatedDate);
```
Comment (mmitche) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Why is the hash comparison necessary here? If the PR or comment was updated, we will get a newer updated date, which should lead to the hash comparison below.  It seems like the old code was correct.  Instead,we now compare twice.

**Comment by mmitche** [comment]
Was the failure case using webhooks?  I didn't see this in my live deployment (which uses webhooks).  I'm wondering whether there is some missing check for closed PRs in the functions calling updatePR

**Comment by DavidTanner** [comment]
@mmitche I did notice that as well.  That is why I haven't merged the code.  I can't figure out why someone might see the plugin going through all of their pull requests.

**Comment by marcmillien** [comment]
Currently downgrading to `1.31.4` solves the issue here.
We are not using webhooks here.

**Comment by mmitche** [comment]
@marcmillien How many open PRs?

**Comment by marcmillien** [comment]
More or less 20, for more or less 3000 closed.

**Comment by mmitche** [comment]
That is super weird, any ideas @DavidTanner?  Maybe the trigger initialization checked for updates on closed PRs?

**Pull request closed without merging**